### ISSUE TITLE
Deriving more traits for Error

### ISSUE THREAD

**Issue by dr-orlovsky** [bitcoindevkit/rust-electrum-client#39]
Currently due to limitation of error data structures from the upstream crates this is impossible. However next weeks I will be fixing `rust-secp256k1`, `rust-bitcoin` & `rust-miniscript` error types to a uniform format, so leaving this issue just not to forget to fix Error derives here once that will have be done.

**Comment by RCasatta** [comment]
@dr-orlovsky is this still an issue?

**Comment by dr-orlovsky** [comment]
Yes, and not all of the upstream PRs required for implementing more derives in rust-electrum are accepted yet (but most is already)
